lib/profile.js

```diff
@@ -44,6 +44,22 @@ module.exports = ({request:client}) => async ({provider, input, output}) => {
     if (provider.flickr) {
       body = JSON.parse(/^.*\((.*)\)/.exec(body)[1])
     }
+    // qq
+    if (provider.qq){
+      options.url = 'https://graph.qq.com/user/get_user_info'
+      body = JSON.parse(/^.*\((.*)\)/.exec(Object.keys(body)[0])[1])
```

Comment:
`https://graph.qq.com/oauth2.0/me?access_token=xxxxxxxxxx` 
The returned content-type is text/html, but the content is actually JSONP
```
callback( {"client_id":"123456","openid":"ABCD1234"} );
```

https://github.com/simov/grant/blob/f1d89d061e7b222afd2966ebadd998e20badbadf/lib/client.js#L41
In this line of code, if the `content-type` is `text/html`, it will try to use `JSON.parser()`, but because the content is JSONP, this will fail, and instead use `qs.parse()`, the return value is `{'callback( {"client_id":"123456","openid":"ABCD1234"} );':''}`,
```node
> var qs = require('qs')
undefined
> let body = 'callback( {"client_id":"123456","openid":"ABCD1234"} );'
undefined
> body = JSON.parse(body)
Uncaught SyntaxError: Unexpected token c in JSON at position 0
> body = qs.parse(body)
{ 'callback( {"client_id":"123456","openid":"ABCD1234"} );': '' }
>
```
This is why need to use `Object.keys(body)[0]` to get the body：
```js
> Object.keys(body)[0]
'callback( {"client_id":"123456","openid":"ABCD1234"} );'
```

![image](https://user-images.githubusercontent.com/7500825/84338335-57a0f380-abce-11ea-8ec0-67d2ceba894b.png)

Reply:
I see, I did the default `qs.parse` in `client.js` because initially Grant was supposed to handle only the token request, but then I decided to re-use that client for the user profile endpoint.